### PR TITLE
Fix [OnServerConnect, OnServerError, OnServerDisconnect] handlers not…

### PR DESCRIPTION
**Pull request by CrHasher**
There were multiple issues with the provided sample:

1. Handlers were not called at all for connect, disconnect, error (Mirror is not issuing expected messages).
2. NetworkServer.Listen expects max connections as parameter not port number.
3. There was a strange Telepathy.dll.meta error when opening the project the first time.
4. Upgraded to latest available Unity 2019.3.11f1.

All other config files omitted and made this commit as clean as possible.

### CODE REVIEW

File: UnityMirror/UnityServer/Assets/Server/Scripts/UnityNetworkServer.cs [hunk 1]
```diff
@@ -57,29 +57,32 @@ private void OnReceiveAuthenticate(NetworkMessage netMsg)
...
-        private void OnServerError(NetworkMessage netMsg)
+        public override void OnServerError(NetworkConnection conn, int errorCode)
         {
+            base.OnServerError(conn, errorCode);
+
             try
             {
-                var error = netMsg.ReadMessage<ErrorMessage>();
-                if (error.value != 0)
+                if (errorCode != 0)
```
Comment (CrHasher) [inline UnityMirror/UnityServer/Assets/Server/Scripts/UnityNetworkServer.cs:83]: Not sure about this check

**Comment by dgkanatsios** [comment]
hey @CrHasher, thanks for the contribution! Is this ready for review?

**Comment by CrHasher** [comment]
> hey @CrHasher, thanks for the contribution! Is this ready for review?

Yes please !

File: UnityMirror/UnityServer/Assets/Server/Scripts/UnityNetworkServer.cs (hunk 1 above)
Comment (dgkanatsios) [reply UnityMirror/UnityServer/Assets/Server/Scripts/UnityNetworkServer.cs:83]: yeah not sure either, can't find a relevant check in the documentation. Can we remove it completely? Also, now that we don't do ReadMessage I don't think we need the try/catch

---
In reply to: [417784333](https://github.com/PlayFab/gsdkSamples/pull/18#discussion_r417784333) [](ancestors = 417784333)

**Comment by dgkanatsios** [comment]
I think we're good, only one comment/question. We can merge after that! Thanks!

File: UnityMirror/UnityServer/Assets/Server/Scripts/UnityNetworkServer.cs (hunk 1 above)
Comment (CrHasher) [reply UnityMirror/UnityServer/Assets/Server/Scripts/UnityNetworkServer.cs]: Changed the code as you suggested.

**Comment by dgkanatsios** [comment]
Thanks for the contribution!

**Comment by CrHasher** [comment]
My pleasure!

**Pull request merged**